barretenberg/cpp/src/barretenberg/dsl/acir_format/honk_recursion_constraint.cpp

```diff
@@ -128,14 +126,85 @@ std::array<uint32_t, HonkRecursionConstraint::AGGREGATION_OBJECT_SIZE> create_ho
                 2 * num_frs_comm) %
                    (num_frs_comm + num_frs_fr * UltraFlavor::BATCHED_RELATION_PARTIAL_LENGTH) ==
                0);
-        vkey->log_circuit_size = (input.proof.size() - HonkRecursionConstraint::inner_public_input_offset -
-                                  UltraFlavor::NUM_WITNESS_ENTITIES * num_frs_comm -
-                                  UltraFlavor::NUM_ALL_ENTITIES * num_frs_fr - 2 * num_frs_comm) /
-                                 (num_frs_comm + num_frs_fr * UltraFlavor::BATCHED_RELATION_PARTIAL_LENGTH);
-        vkey->circuit_size = (1 << vkey->log_circuit_size);
-        vkey->num_public_inputs = input.public_inputs.size();
-        vkey->pub_inputs_offset = UltraFlavor::has_zero_row ? 1 : 0;
+        auto log_circuit_size = (input.proof.size() - HonkRecursionConstraint::inner_public_input_offset -
```

Comment:
this formula here just always computes CONST_PROOF_SIZE_LOG_N now that the proof size is constant.

Reply:
Added a note. Leaving the formula because its complicated and we;'ll almost certainly revert all of this const proof size nonsense